### ISSUE TITLE
Investigate Bluemix sqldb service shutdown

### ISSUE THREAD

**Issue by tdalbo92** [watson-developer-cloud/questions-with-classifier-ega#68]
We need to determine what changes, if any, need to be made in order to use a suitable replacement db service on Bluemix.

**Comment by yelkurdi** [comment]
Hi, 
Any plans you have for this issue?
Yousef

**Comment by tdalbo92** [comment]
In progress.  This codebase is kept alive by me working in my spare time outside of work, so Pull Requests are always appreciated.

My goal is to have an alternative Bluemix db working before the April 4th shutdown.

**Comment by suhel1982** [comment]
Hi Tom- Did you make any progress in this ? I was poking around with this application a few days back and realized that the SQL DB is deprecated. 

Please advise if you have any other working solution. 
Thanks!

**Comment by tdalbo92** [comment]
Unfortunately, we didn't have the time to dedicate resources to fixing this :(

**Comment by sarielda** [comment]
I'm trying to migrate from SQL DB to dashDB. Do you happen to know the code piece (or files) that sets up the database tables? I tried to locate it but couldn't find it easily.

**Comment by tdalbo92** [comment]
Yeah, so sadly the reason why I stopped pursing this is because my JPA knowledge is less than great.  

In theory all you need to do is all the appropriate JARs for the JPA layers for whatever database you want to use.  Then you'd need to modify persistence.xml and potentially web.xml, and that should be it.  

However...  that's all in theory.  My knowledge ends right around there.

**Comment by sarielda** [comment]
I managed to switch from SQL DB to dashDB (also available on Bluemix). Both databases are based on DB2 with some differences that need to apply on the code. Here's what I did:
(1) Add a service of type dashDB to the app. Choose the entry plan
(2) Update the manifest file to include the dashDB details instead of SQLDB
(3) Create the database tables manually. The app code originally uses jpa SynchronizeMappings to create the tables. dashDB has an issue with CLOB data types since it defaults its tables to ORGANIZE BY COLUMNS. I couldn't find a way to reflect that on the jpa Entity definition but i'll give it a shot later on. In the meanwhile it's straight forward to create the app tables on dashDB using the attached DDL file.

The rest is as described in the repo README. I'll try to clean up my code and send a pull request shortly.
[Script.txt](https://github.com/watson-developer-cloud/questions-with-classifier-ega/files/280524/Script.txt)

**Comment by tdalbo92** [comment]
Incredible work!  Thank you so much, I wasn't able to get past the "Create db service in Bluemix" step :D 

You've kept this project alive!
